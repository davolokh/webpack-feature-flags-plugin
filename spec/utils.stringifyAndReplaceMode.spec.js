const {
  stringifyAndReplaceMode,
  PREDEFINED_MODE_VALUES,
} = require('../src/utils');
const stubConfig = require('./webpack.feature-flags.sample.config');


describe('stringifyAndReplaceMode', () => {
  it('stub data, with PROD mode', () => {
    expect(stringifyAndReplaceMode(stubConfig, PREDEFINED_MODE_VALUES, 'PROD')).toMatchSnapshot();
  });

  it('stub data, with DEV mode', () => {
    expect(stringifyAndReplaceMode(stubConfig, PREDEFINED_MODE_VALUES, 'DEV')).toMatchSnapshot();
  });

  it('stub data, with added USER_EXTENDED_MODE mode', () => {
    expect(stringifyAndReplaceMode(stubConfig, [...PREDEFINED_MODE_VALUES, 'USER_EXTENDED_MODE'], 'USER_EXTENDED_MODE')).toMatchSnapshot();
  });

  it('stub data, with UNKNOWN_MODE mode', () => {
    expect(stringifyAndReplaceMode(stubConfig, PREDEFINED_MODE_VALUES, 'UNKNOWN_MODE')).toMatchSnapshot();
  });
  
});
