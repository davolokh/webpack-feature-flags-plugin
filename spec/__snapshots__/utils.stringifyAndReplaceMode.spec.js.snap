// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringifyAndReplaceMode stub data, with DEV mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "false",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "true",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;

exports[`stringifyAndReplaceMode stub data, with PROD mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "true",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "false",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;

exports[`stringifyAndReplaceMode stub data, with UNKNOWN_MODE mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "false",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "false",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;

exports[`stringifyAndReplaceMode stub data, with added USER_EXTENDED_MODE mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "false",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "true",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "false",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;
